java/core/src/java/org/apache/orc/impl/ParserUtils.java

```diff
@@ -459,10 +459,31 @@ public static ColumnVector[] findColumnVectors(TypeDescription schema,
                                                  boolean isCaseSensitive,
                                                  VectorizedRowBatch batch) {
     List<String> names = ParserUtils.splitName(source);
-    ColumnFinder result = new ColumnFinder(schema, batch, names.size());
+    ColumnFinder result = new ColumnFinder(schema, removeAcidColumnsFromVectorizedRowBatch(schema, batch), names.size());
     findColumn(removeAcid(schema), names, isCaseSensitive, result);
     return result.result;
   }
+  
+  private static VectorizedRowBatch removeAcidColumnsFromVectorizedRowBatch(TypeDescription schema, VectorizedRowBatch batch) {
+    final int acidColumnOffset = 5;
+    if(SchemaEvolution.checkAcidSchema(schema)) {
+      VectorizedRowBatch vectorizedRowBatch = new VectorizedRowBatch(batch.cols.length - acidColumnOffset, batch.size);
+      System.arraycopy(batch.selected, 0, vectorizedRowBatch.selected, 0, vectorizedRowBatch.selected.length);
+      System.arraycopy(batch.projectedColumns, acidColumnOffset, vectorizedRowBatch.projectedColumns, 0, batch.projectedColumns.length - acidColumnOffset);
+      vectorizedRowBatch.setPartitionInfo(batch.getDataColumnCount(), batch.getPartitionColumnCount());
+      vectorizedRowBatch.selectedInUse = batch.selectedInUse;
+      vectorizedRowBatch.endOfFile = batch.endOfFile;
+
+      StructColumnVector structColumnVector = (StructColumnVector)(batch.cols[acidColumnOffset]);
+      ColumnVector[] columnVectors = new ColumnVector[structColumnVector.fields.length];
+      for (int i = 0; i < columnVectors.length; i++) {
+        columnVectors[i] = structColumnVector.fields[i];
+      }
+      vectorizedRowBatch.cols = columnVectors;
+      return vectorizedRowBatch;
+    }
+    return batch;
```

Comment:
I might be missing the logic of where we are updating back the changes that are not referenced example batch.size back to the batch after the filter processing is complete.

I go back to the earlier comment that we should do this differently.

Reply:
Changed the logic. Please check the PR.